bitcoin/src/util/hash.rs

```diff
@@ -39,6 +39,44 @@ where
 
 /// Calculates the merkle root of an iterator of *hashes*.
 ///
+/// This fcuntion calculate the root from a merkle path when we need to derive the root from the
+/// txs list [bitcoin_merkle_root] should be used.
+///
+/// # Returns
+/// - `None` if `hashes` is empty. The merkle root of an empty tree of hashes is undefined.
+/// - `Some(hash)` if `hashes` contains one element. A single hash is by definition the merkle root.
+/// - `Some(merkle_root)` if length of `hashes` is greater than one.
+pub fn bitcoin_merkle_root_from_path<T, I>(mut hashes: I) -> Option<T>
+where
+    T: Hash + Encodable,
+    <T as Hash>::Engine: io::Write,
```

Comment:
Is there any use of non-sha256 function? If not I'd just hard-code SHA256.

Reply:
Do we need arbitrary function or do we want just sha256?

Reply:
Both transaction merkle tree and taproot use SHA256. We don't have other merkle trees in bitcoin.

Reply:
Taproot tree uses a single tagged sha256. Transaction merkle tree uses double-sha256. So these are actually different hashes. I'm not sure about the segwit witness merkle tree.

Reply:
Ah, good point. OK, let's have them generic.